js_modules/dagster-ui/packages/ui-core/src/runs/RunsFeedRoot.tsx

```diff
@@ -50,49 +49,33 @@ export const RunsFeedRoot = () => {
 
   const [filterTokens, setFilterTokens] = useQueryPersistedRunFilters();
   const filter = runsFilterForSearchTokens(filterTokens);
-  const [statusTokens, nonStatusTokens] = useMemo(
-    () => partition(filterTokens, (token) => token.token === 'status'),
-    [filterTokens],
-  );
 
   const [view, setView] = useQueryPersistedState<RunsFeedView>({
     encode: (v) => ({view: v && v !== RunsFeedView.ROOTS ? v.toLowerCase() : undefined}),
     decode: (qs) => (qs.view || RunsFeedView.ROOTS).toUpperCase(),
   });
 
   const currentTab = useSelectedRunsFeedTab(filterTokens, view);
-  const currentTabSpecifiesStatuses = !['all', 'backfills'].includes(currentTab);
 
   const setFilterTokensWithStatus = useCallback(
     (tokens: RunFilterToken[]) => {
-      if (currentTabSpecifiesStatuses) {
-        setFilterTokens([...statusTokens, ...tokens]);
-      } else {
-        setFilterTokens(tokens);
-      }
+      setFilterTokens(tokens);
     },
-    [setFilterTokens, currentTabSpecifiesStatuses, statusTokens],
+    [setFilterTokens],
   );
 
   const onAddTag = useCallback(
     (token: RunFilterToken) => {
       const tokenAsString = tokenToString(token);
-      if (!nonStatusTokens.some((token) => tokenToString(token) === tokenAsString)) {
-        setFilterTokensWithStatus([...nonStatusTokens, token]);
+      if (!filterTokens.some((token) => tokenToString(token) === tokenAsString)) {
+        setFilterTokensWithStatus([...filterTokens, token]);
```

Comment:
Not that this changed in this PR, but reuse of `token` as the argument name in the nested function is slightly confusing here.